src/com/projectkorra/projectkorra/ability/AirAbility.java

```diff
@@ -106,28 +106,61 @@ public static boolean isWithinAirShield(final Location loc) {
 
 	/**
 	 * Plays an integer amount of air particles in a location.
+	 * Deprecated. Use {@link AirAbility#playAirbendingParticles(CoreAbility, Location, int)} instead
 	 *
 	 * @param loc The location to use
 	 * @param amount The amount of particles
 	 */
+	@Deprecated
 	public static void playAirbendingParticles(final Location loc, final int amount) {
 		playAirbendingParticles(loc, amount, Math.random(), Math.random(), Math.random());
 	}
 
+
+	/**
+	 * Plays an integer amount of air particles in a location.
+	 *
+	 * @param ability The ability this particle is spawned for
+	 * @param loc The location to use
+	 * @param amount The amount of particles
+	 */
+	public static void playAirbendingParticles(final CoreAbility ability, final Location loc, final int amount) {
+		playAirbendingParticles(ability, loc, amount, Math.random(), Math.random(), Math.random());
+	}
+
+
 	/**
 	 * Plays an integer amount of air particles in a location with a given
 	 * xOffset, yOffset, and zOffset.
+	 * Deprecated. Use {@link AirAbility#playAirbendingParticles(CoreAbility, Location, int, double, double, double)} instead
 	 *
 	 * @param loc The location to use
 	 * @param amount The amount of particles
 	 * @param xOffset The xOffset to use
 	 * @param yOffset The yOffset to use
 	 * @param zOffset The zOffset to use
 	 */
+	@Deprecated
 	public static void playAirbendingParticles(final Location loc, final int amount, final double xOffset, final double yOffset, final double zOffset) {
 		getAirbendingParticles().display(loc, amount, xOffset, yOffset, zOffset);
 	}
 
+	/**
+	 * Plays an integer amount of air particles in a location with a given
+	 * xOffset, yOffset, and zOffset.
+	 *
+	 * @param ability The ability this particle is spawned for
+	 * @param loc The location to use
+	 * @param amount The amount of particles
+	 * @param xOffset The xOffset to use
+	 * @param yOffset The yOffset to use
+	 * @param zOffset The zOffset to use
+	 */
+	public static void playAirbendingParticles(final CoreAbility ability, final Location loc, final int amount, final double xOffset, final double yOffset, final double zOffset) {
+		getAirbendingParticles().display(loc, amount, xOffset, yOffset, zOffset);
```

Comment:
These changes are a little redundant. You don't use that CoreAbility parameter anywhere in the subsequent code despite deprecating the old functions. 

Is there a motivation behind this?

Reply:
It would be for futute use, further explanation can be found in my reply to Christopher, where I explain what it could be used for. You can also ask Drieg_Michihi on discord, as this change would be also beneficial for him.

playFirebendingParticle has gone the same path (checking for blue fire), I have done the same will all elements, but in a less destructive way. 
I completely understand this is quite a large change, but all previous code will still function. The faster this is done, the better.